src/main/java/com/example/Ecommerce/service/CartItemQueueService.java

```diff
@@ -0,0 +1,80 @@
+package com.example.Ecommerce.service;
+
+import com.example.Ecommerce.entity.*;
+import com.example.Ecommerce.exception.ResourceNotFoundException;
+import com.example.Ecommerce.model.QueueItem;
```

Comment:
QueueItem could be an inner class here, it make sense to be tightly coupled with CartItemQueueService